### PR TITLE
Clean up code for fire extinguishers and water balloons

### LINKED ISSUE #24159: Can't fill a water ballon with a sink
You wash it instead. Ruined my immulsions

[easy]

### PR DESCRIPTION
**Pull request by ghost**
Went ahead and just cleaned up everything because holy fucking cow. Also some bonus changes

- Mini extinguisher can no longer cram items in its nozzle (it's a tiny ass pocket fire extinguisher, the big version can already only cram tiny items in)
- Can now fill fire extinguishers and water balloons at a sink. Will no longer wash them. Fixes #24159
- Water balloon now audibly explodes when it impacts something (same sound as the AME, it fits, trust me)
- Formatting, visible messages, all the good stuff

:cl:
 * rscadd: You can now fill fire extinguishers and water balloons in sinks
 * rscdel: You can no longer cram items into mini extinguishers
 * tweak: Water balloons now explode audibly, to make your sulphuric acid ballons have more 'oompf' to them

### CODE REVIEW

File: code/game/objects/items/weapons/extinguisher.dm [hunk 1]
```diff
@@ -41,124 +50,136 @@
...
-			to_chat(user, "You cram \the [W] into the nozzle of \the [src].")
+			to_chat(user, "<span class='notice'>You cram [W] into [src]'s nozzle.</span>")
 			message_admins("[user]/[user.ckey] has crammed \a [W] into a [src].")
 
 /obj/item/weapon/extinguisher/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
 	if(proximity_flag)
-		if((istype(target, /obj/structure/reagent_dispensers)))
-			target.reagents.trans_to(src, 50, log_transfer = TRUE, whodunnit = user)
-			to_chat(user, "<span class='notice'>\The [src] is now refilled</span>")
+		if(istype(target, /obj/structure/reagent_dispensers/watertank))
```
Comment (Shadowmech88) [inline code/game/objects/items/weapons/extinguisher.dm]: Doesn't this mean extinguishers can no longer be filled with welding fuel from welding tanks?
Why?

**Comment by Shadowmech88** [comment]
Why not base filling from sinks on whether the extinguisher has been wrenched open, same as filling from something like a beaker?
What if you have a dirty fire extinguisher and you want to wash it?

File: code/game/objects/items/toys.dm [hunk 2]
```diff
@@ -29,58 +29,72 @@
...
+					to_chat(user, "<span class='danger'>The acid chews through the balloon!</span>")
 					O.reagents.reaction(user)
 					qdel(src)
 					return
 				else
-					src.desc = "A translucent balloon with some form of liquid sloshing around in it."
-					to_chat(user, "<span class = 'info'>You fill the balloon with the contents of \the [O].</span>")
-					O.reagents.trans_to(src, 10)
-	src.update_icon()
+					user.visible_message("<span class='notice'>[user] fills [src] using [O]</span>",
```
Comment (ShiftyRail) [inline code/game/objects/items/toys.dm]: Add a full stop in here, please.

File: code/game/objects/items/weapons/extinguisher.dm (hunk 1 above)
Comment (ghost) [reply code/game/objects/items/weapons/extinguisher.dm]: Snap, correcting that

**Comment by ghost** [comment]
@Shadowmech88 Will not work because I had to add an exception to washing in the sink

**Comment by abernd** [comment]
Can't hide the nuclear disk in an empty mini extinguisher in a toolbelt anymore, stealth nuke ops buff.

**Comment by Shadowmech88** [comment]
@dylanstrategie 
Couldn't you just check for `istype(O, /obj/item/weapon/extinguisher) && !(O.flags & OPENCONTAINER)` in sink code instead of checking for just `istype(O, /obj/item/weapon/extinguisher)`?

**Comment by SonixApache** [comment]
CLANG sound for waterballoon impact is dumb
make washing an intent thing so i can wash my dirty extinguishers pls

File: code/game/objects/items/toys.dm [hunk 3]
```diff
@@ -29,58 +29,72 @@
...
 	return
 
-/obj/item/toy/waterballoon/afterattack(atom/A as mob|obj, mob/user as mob)
-	if (istype(A, /obj/structure/reagent_dispensers/watertank) && get_dist(src,A) <= 1)
-		A.reagents.trans_to(src, 10)
-		to_chat(user, "<span class = 'notice'>You fill the balloon with the contents of \the [A].</span>")
-		src.desc = "A translucent balloon with some form of liquid sloshing around in it."
-		src.update_icon()
-	return
+/obj/item/toy/waterballoon/afterattack(atom/A, mob/user, proximity_flag, click_parameters)
```
Comment (Kurfursten) [inline code/game/objects/items/toys.dm:49]: why all this? why not just make it an opencontainer?

that would solve your examine issue AND let you fill it in any way of your choice just like any other containers, you could do away with basically all of the examine(), attackby() and afterattack()

Comment (ghost) [reply code/game/objects/items/toys.dm:49]: Well... I don't know how to is why. Oof, reagent code

Comment (Kurfursten) [reply code/game/objects/items/toys.dm:49]: Just delete all of those 3 procs and give it the flag OPENCONTAINER. If you need help I'll be available on discord when I get home from thanksgiving.

**Comment by MadmanMartian** [comment]
wow, they're right, your format changes really do make reviewing a pain

**Comment by jknpj** [comment]
dylan pls

**Comment by jknpj** [comment]
Ping me if you want to fix/reopen this.

**Pull request closed without merging**